src/cli/migrate/nodeVersion/checks.ts

```diff
@@ -0,0 +1,111 @@
+import findUp from 'find-up';
+import fs from 'fs-extra';
+import { coerce, lte, satisfies } from 'semver';
+import { type ZodRawShape, z } from 'zod';
+
+import { log } from '../../../utils/logging';
+
+const getParentFile = async (file: string) => {
+  const path = await findUp(file, { cwd: process.cwd() });
+  if (!path) {
+    throw new Error(`${file} not found`);
+  }
+  return fs.readFile(path, 'utf-8');
+};
+
+export const extractFromParentPackageJson = async <T extends ZodRawShape>(
+  schema: z.ZodObject<T>,
+): Promise<z.infer<typeof schema> | undefined> => {
+  const packageJson = await getParentFile('package.json');
+  let rawJSON;
+  try {
+    rawJSON = JSON.parse(packageJson) as unknown;
+  } catch {
+    throw new Error('package.json is not valid JSON');
+  }
+  const result = schema.safeParse(rawJSON);
+  if (!result.success) {
+    return undefined;
+  }
+
+  return result.data;
+};
+
+export const isPatchableServerlessVersion = async (): Promise<boolean> => {
+  const serverlessVersion = (
+    await extractFromParentPackageJson(
+      z.object({
+        devDependencies: z.object({
+          serverless: z.string(),
+        }),
+      }),
+    )
+  )?.devDependencies.serverless;
+
+  if (!serverlessVersion) {
+    log.subtle('Serverless version not found, assuming it is not a dependency');
```

Comment:
Would it be useful to log the `package.json` that we found, if any?

```suggestion
    log.subtle(`Serverless version not found in ${packageJsonRelativePath}, assuming it is not a dependency`);
```